src/librustc_metadata/encoder.rs

```diff
@@ -267,7 +276,7 @@ fn encode_enum_variant_info<'a, 'tcx>(ecx: &EncodeContext<'a, 'tcx>,
             }
         }
 
-        index.record(vid, rbml_w);
+        let _task = index.record(vid, rbml_w);
```

Comment:
Can you change that to take a closure. It would make it more obvious that we are working within a DepTask.